examples/erc20-counter/apps/src/bin/publisher.rs

```diff
@@ -66,18 +75,24 @@ struct Args {
     account: Address,
 }
 
-fn main() -> Result<()> {
+#[tokio::main]
+async fn main() -> Result<()> {
     // Initialize tracing. In order to view logs, run `RUST_LOG=info cargo run`
     tracing_subscriber::fmt()
         .with_env_filter(EnvFilter::from_default_env())
         .init();
-
-    // parse the command line arguments
+    // Parse the command line arguments.
     let args = Args::parse();
 
-    // Create an EVM environment from an RPC endpoint and a block number. If no block number is
-    // provided, the latest block is used.
-    let mut env = EthEvmEnv::from_rpc(&args.rpc_url, None)?;
+    // Create an alloy provider for that private key and URL.
+    let wallet = EthereumWallet::from(args.eth_wallet_private_key);
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet)
+        .on_http(args.rpc_url);
```

Comment:
Very interesting, I didn't even know of this one. It even takes a string similar to the eth provider before...

 🤔 I kinda liked the change to Url for its additional type safety and I guess HTTP is the more realistic provider as we don't rely on any pubsub and even then we have the `from_provider` for any alloy provider...

Reply:
On the question of using `Url`, I also like that from a semantics point of view. With `on_builtin`, we could call `url.as_str` to pass it in, so we can keep that change either way. On the use of HTTP provider. I do see that we constrain the types to HTTP providers anyway. Something we should think about before marking 1.0 is whether we should/can loosen thew type constraints there. It's really not a major issue in any case.

Reply:
The types of `EthEvmEnv::from_rpc` are only `Http` because this is what `.on_http` returns. There is also https://github.com/risc0/risc0-ethereum/blob/eded136a25ff400e3f108da9ad17ae51e4c86fbd/steel/src/host/mod.rs#L62
which works with any generic alloy provider.